Issue with ES >= v7.15.0: TypeError: search() got multiple values for keyword argument 'size'

<!-- Bug report -->

**Elasticsearch version** : 7.15.1

**`elasticsearch-py` version (`elasticsearch.__versionstr__`)**: 7.15.1

Please make sure the major version matches the Elasticsearch server you are running.

**Description of the problem including expected versus actual behavior**:

Hello,

I'm upgrading ES from v7.14.2 to v7.15.1 and I'm getting an error on code that used to work fine:

```
resp = client.search(
            scroll=scroll, size=size, request_timeout=request_timeout, **search_kwargs
        )
E       TypeError: search() got multiple values for keyword argument 'size'

/usr/local/lib/python3.8/dist-packages/elasticsearch/helpers/actions.py:571: TypeError
```

I'm doing a search.scan() without any extra parameters, and the search itself has from=0 and size=0.
If I remove the pagination entirely, it works.
Any idea what might be going on?
Downgrading ES to v7.14.2 solves the issue.
I've checked recent bug reports on elasticsearch and elasticsearch-dsl but nothing similar shows up.
The changelog of elasticsearch itself also doesn't give any clues.

Greetings,

Jan

**Steps to reproduce**:

--- a/elasticsearch/_async/helpers.py
+++ b/elasticsearch/_async/helpers.py
@@ -355,16 +355,19 @@ async def async_scan(
     )
     client._client_meta = (("h", "s"),)
 
-    # initial search
-    search_kwargs = query.copy() if query else {}
-    search_kwargs.update(kwargs)
-    search_kwargs["scroll"] = scroll
-    search_kwargs["size"] = size
-
     try:
+        search_kwargs = query.copy() if query else {}
+        search_kwargs.update(kwargs)
+        search_kwargs["scroll"] = scroll
+        search_kwargs["size"] = size
         resp = await client.search(**search_kwargs)
+
+    # Try the old deprecated way if we fail immediately on parameters.
     except TypeError:
-        resp = await client.search(body=query, scroll=scroll, size=size, **kwargs)
+        search_kwargs = kwargs.copy()
+        search_kwargs["scroll"] = scroll
+        search_kwargs["size"] = size
+        resp = await client.search(body=query, **search_kwargs)
 
     scroll_id = resp.raw.get("_scroll_id")
     scroll_transport_kwargs = pop_transport_kwargs(scroll_kwargs)

--- a/elasticsearch/helpers/actions.py
+++ b/elasticsearch/helpers/actions.py
@@ -566,15 +566,20 @@ def scan(
     )
     client._client_meta = (("h", "s"),)
 
-    # initial search
-    search_kwargs = query.copy() if query else {}
-    search_kwargs.update(kwargs)
-    search_kwargs["scroll"] = scroll
-    search_kwargs["size"] = size
     try:
+        search_kwargs = query.copy() if query else {}
+        search_kwargs.update(kwargs)
+        search_kwargs["scroll"] = scroll
+        search_kwargs["size"] = size
         resp = client.search(**search_kwargs)
+
+    # Try the old deprecated way if we fail immediately on parameters.
     except TypeError:
-        resp = client.search(body=query, scroll=scroll, size=size, **kwargs)
+        search_kwargs = kwargs.copy()
+        search_kwargs["scroll"] = scroll
+        search_kwargs["size"] = size
+        resp = client.search(body=query, **search_kwargs)
+
     scroll_id = resp.raw.get("_scroll_id")
     scroll_transport_kwargs = pop_transport_kwargs(scroll_kwargs)
     if scroll_transport_kwargs:

--- a/test_elasticsearch/test_async/test_server/test_rest_api_spec.py
+++ b/test_elasticsearch/test_async/test_server/test_rest_api_spec.py
@@ -147,7 +147,7 @@ class AsyncYamlRunner(YamlRunner):
         warnings.simplefilter("always", category=ElasticsearchWarning)
         with warnings.catch_warnings(record=True) as caught_warnings:
             try:
-                self.last_response = await api(**args).raw
+                self.last_response = (await api(**args)).raw
             except Exception as e:
                 if not catch:
                     raise

